build-tools/src/main/java/org/elasticsearch/gradle/plugin/PluginBuildPlugin.java

```diff
@@ -114,64 +116,41 @@ private static void configureDependencies(final Project project) {
-                        }
-                    }
-                });
-            }
-        });
-
-        final var buildProperties = project.getTasks().register("pluginProperties", Copy.class, copy -> {
-            copy.dependsOn(copyPluginPropertiesTemplate);
-            copy.from(templateFile);
-            copy.into(new File(project.getBuildDir(), "generated-resources"));
-
-            if (extension.getName() == null) {
-                throw new InvalidUserDataException("name is a required setting for esplugin");
-            }
-
-            if (extension.getDescription() == null) {
-                throw new InvalidUserDataException("description is a required setting for esplugin");
-            }
-
-            if (extension.getType().equals(PluginType.BOOTSTRAP) == false && extension.getClassname() == null) {
-                throw new InvalidUserDataException("classname is a required setting for esplugin");
-            }
 
-            Map<String, Object> map = new LinkedHashMap<>(12);
-            map.put("name", extension.getName());
-            map.put("description", extension.getDescription());
-            map.put("version", extension.getVersion());
-            map.put("elasticsearchVersion", Version.fromString(VersionProperties.getElasticsearch()).toString());
-            map.put("javaVersion", project.getExtensions().getByType(JavaPluginExtension.class).getTargetCompatibility().toString());
-            map.put("classname", extension.getType().equals(PluginType.BOOTSTRAP) ? "" : extension.getClassname());
-            map.put("extendedPlugins", extension.getExtendedPlugins().stream().collect(Collectors.joining(",")));
-            map.put("hasNativeController", extension.isHasNativeController());
-            map.put("requiresKeystore", extension.isRequiresKeystore());
-            map.put("type", extension.getType().toString());
-            map.put("javaOpts", extension.getJavaOpts());
-            map.put("licensed", extension.isLicensed());
-
-            copy.expand(map);
-            copy.getInputs().properties(map);
+        final var buildProperties = project.getTasks().register("pluginProperties", GeneratePluginPropertiesTask.class, task -> {
+            /*task.doFirst(t -> {
+                if (extension.getName() == null) {
+                    throw new InvalidUserDataException("name is a required setting for esplugin");
+                }
+
+                if (extension.getDescription() == null) {
+                    throw new InvalidUserDataException("description is a required setting for esplugin");
+                }
+
+                if (extension.getType().equals(PluginType.BOOTSTRAP) == false && extension.getClassname() == null) {
+                    throw new InvalidUserDataException("classname is a required setting for esplugin");
+                }
+            });*/
+            task.getPluginName().set(providerFactory.provider(extension::getName));
```

Comment:
We could make this cleaner by finally porting the plugin extension to use  lazy properties. But we should do this in a dedicated PR. I've created https://github.com/elastic/elasticsearch/issues/86057 to track this